### PR TITLE
feat: refactor fragment helper in react ot handle keys

### PR DESCRIPTION
**Pull request by nmerget**
## Description

Please provide the following information:

- Update `helper.ts` inside `react` target to properly use `Fragment key="xxx"`

Make sure to follow the PR preparation steps in [CONTRIBUTING.md](../CONTRIBUTING.md#preparing-your-pr) before submitting your PR:

- [x] format the codebase: from the root, run `yarn fmt:prettier`.
- [ ] update all snapshots (in core & CLI): from the root, run `yarn test:update`
- [x] add Changeset entry: from the root, run `yarn g:changeset` and follow the CLI instructions. Alternatively, use the Changeset Github Bot to create the file.

### CODE REVIEW

File: packages/core/src/generators/react/helpers.ts [hunk 1]
```diff
@@ -21,9 +21,18 @@ export const processBinding = (str: string, options: ToReactOptions) => {
 
 export const openFrag = (options: ToReactOptions) => getFragment('open', options);
 export const closeFrag = (options: ToReactOptions) => getFragment('close', options);
-export function getFragment(type: 'open' | 'close', options: ToReactOptions) {
-  const tagName = options.preact ? 'Fragment' : '';
-  return type === 'open' ? `<${tagName}>` : `</${tagName}>`;
+export function getFragment(type: 'open' | 'close', options: ToReactOptions, node?: MitosisNode) {
+  let tag = '';
+  if (node?.name === 'Fragment' && node.bindings['key']) {
+    // TODO: Is there another output which cannot use "React.Fragment"??
```
Comment (nmerget) [inline packages/core/src/generators/react/helpers.ts:27]: @samijaber I am not sure if `rsc` or `react-native` has another Fragment syntax. Do you know more about it?

**Comment by nmerget** [comment]
image  <- thats a sign :D

File: packages/core/src/generators/react/helpers.ts (hunk 1 above)
Comment (samijaber) [reply packages/core/src/generators/react/helpers.ts:27]: No, all React formats (normal, rsc, react native) can use the same `React.Fragment` syntax

**Review by samijaber — APPROVED** [review]
sweet!

File: packages/core/src/__tests__/__snapshots__/react-native.test.ts.snap [hunk 2]
```diff
@@ -10025,13 +10101,13 @@ type Props = {
...
-    console.log(\\"props.foo.value.bar.baz changed\\", props.foo.bar.baz);
-  }, [props.id, props.foo.bar.baz]);
+    console.log(\\"props.foo.value.bar.baz changed\\", props.foo.value.bar.baz);
+  }, [props.id, props.foo.value.bar.baz]);
 
   return (
     <View style={styles.view1}>
       <Text>{props.id}</Text>
-      <Text>{props.foo.bar.baz}</Text>
+      <Text>{props.foo.value.bar.baz}</Text>
```
[the remark this answers is not in the document]
Comment (nmerget) [reply packages/core/src/__tests__/__snapshots__/react-native.test.ts.snap]: updated them with the output from `test-update` job ;)

**Comment by samijaber** [comment]
@nmerget this is ready to merge, but your snapshots are still out of date 😬 . might be due to a recent merge?

**Comment by nmerget** [comment]
> @nmerget this is ready to merge, but your snapshots are still out of date 😬 . might be due to a recent merge on `main`?

fixed ^.^

**Pull request merged**